src/contract.rs

```diff
@@ -1,66 +1,122 @@
+use std::env;
+
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
-use cosmwasm_std::{DepsMut, Env, MessageInfo, Response, StdError, StdResult};
+use cosmwasm_std::{
+    BankMsg, Coin, CosmosMsg, DepsMut, Env, MessageInfo, Response, StdError, StdResult,
+};
 
 use cw2::set_contract_version;
 
 use crate::error::ContractError;
 use crate::msg::{ExecuteMsg, InstantiateMsg};
-use crate::state::CONFIG;
+use crate::state::{Owner, CONFIG};
 
 // version info for migration info
 const CONTRACT_NAME: &str = "crates.io:disburser";
 const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
+// Define the `instantiate` function that takes in several arguments and returns a `StdResult<Response>`.
 pub fn instantiate(
     deps: DepsMut,
     _env: Env,
     _info: MessageInfo,
     msg: InstantiateMsg,
 ) -> StdResult<Response> {
+    // Set the contract version.
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
 
+    let mut seen_addresses: Vec<String> = vec![];
     let mut total_ownership: u8 = 0;
 
-    // Check that the sum of the individual ownerships is equal to 100
     for owner in &msg.owners {
         total_ownership += owner.ownership;
+
+        if seen_addresses.contains(&owner.address.to_string()) {
+            return Err(StdError::generic_err(
+                "Duplicate owner address has been input more than once",
+            ));
+        } else {
+            seen_addresses.push(owner.address.to_string());
+        }
     }
 
+    // Check if the total ownership is equal to 100%. If not, return an error.
     if total_ownership != 100 {
         return Err(StdError::generic_err("Total Ownership must equal 100%."));
     }
 
+    // Check if each owner's individual ownership is greater than 0%. If not, return an error.
     for owner in &msg.owners {
         if owner.ownership == 0 {
             return Err(StdError::generic_err(
-                "Individual Ownership must be greater than 0",
+                "Individual Ownership must be greater than 0.",
             ));
         }
     }
```

Comment:
1. Why you do 2 loops over the same thing? just combine them into one.
2. Why you using generic error? use your custom errors instead.

Reply:
1. duh... thank you!
2. I'm using generic errors because it's simple, and it works nicely. If it isn't broke, don't fix it at this stage I guess, but having custom errors is probably better. I will change these eventually.